README.md

```diff
@@ -13,10 +13,147 @@ pip install -e ".[dev]"
+"on-the-fly" as arguments to `main.py`, for example
+
+```bash
+python main.py pretrain.epochs=20
+```
+
+overwrites the `pretrain.epochs` parameter. Notice that `pretrain.epochs` was specified to ArgParse in advance.  
+The wrapper specifies `"configs"` as the config path. This is important because it tells `hydra` where the root 
+directory is, which allows configs to reference other configs within, among other things. 
+
+#### 2. Inheritance
+
+You can build a config file through inheritance of another to override certain params, or reference another file to 
+incorporate it as a structure. Consider dir structure like the following, and then the `mnist.yaml` config.
+```text
+├── configs
+│   ├── base_config.yaml
+│   ├── mnist.yaml
+│   ├── finetune
+│   │   ├── fashion.yaml
+│   ├── pretrain
+│   │   ├── mnist.yaml
+│   ├── unadapt
+│   │   ├── gradient.yaml
+└── main.py
+```
+```yaml
+# mnist.yaml
+defaults:
+  - base_config
+  # Notice the keywords below match config dirs
+  - pretrain: mnist
+  - unadapt: gradient
+  - /finetune/fashion
+  - _self_
+
+# Base name identifying the run
+basename: mnist  
+```
+- `defaults:` is a special param that `hydra` picks up for importing configs
+- First we reference the `base_config.yaml` that provides some params like `seed: 0` etc.
+- Then we import three other config files as additional params.
+- Finally, we overwrite `basename`, which would normally be `basename: default` from `base_config.yaml`
+
+The final config would look like
+```yaml
+seed: 0
+model: MLP
+pretrained: false
+basename: mnist
+pretrain:
+  dataset: MNIST
+  epochs: 8
+  lr: 0.001
+  batch_size: 128
+  test_batch_size: 1000
+unadapt:
+  method: gradient
+  loss: fim
+  reduce: trace_max
+  lam: 0.1
+  lr: 0.001
+  epochs: 1
+finetune:
+  dataset: FashionMNIST
+  epochs: 1
+  lr: 0.001
+  batch_size: 128
+  test_batch_size: 1000
+```
+Notice how the three `pretrain/unadapt/finetune` structs are automatically named. This actually _isn't_ because we 
+wrote `pretrain: ...`, but is because of the configs directory structure (`pretrain:` just tells hydra to look in 
+the `configs/pretrain` directory for `"mnist.yaml"`). I also include `/finetune/fashion` to further demonstrate this 
+directory-lookup behavior.
+
+#### 3. Overrides at runtime to `main.py`
+
+If we just rain
+```bash
+python main.py
+```
+we would load the default `base_config.yaml` (because that is specified in the `hydra.main` wrapper).  
+To work from a different config file we call
+```bash
+python main.py --config-name=mnist
+```
+This will be found because `mnist.yaml` is in the `configs` root. We can override any of the parameters loaded from 
+the command line
+```bash
+python main.py --config-name=mnist pretrain.epochs=20 seed=8675309
+# Selecting a different unadapt method
+python main.py --config-name=mnist unadapt=prune
+```
+You can also provide one-off params with `+`:
+```bash
+python main.py --config-name=mnist +slurm_log=/slurm/path
+```
```

Comment:
Are we ever gonna use one-off params? Possibly better to just require everything be defined in the config already, then we don't have to check if things are in the config first. Maybe the exception to this could be boolean flags we want to easily enable, like `+disablewandb`, but you could also just have that explicit in the configs as `wandb: enabled` and override it with `wandb=disabled`

Reply:
I was just describing built-in hydra functionality. We can't stop users from doing it, but i'll remove